lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * WP_Theme_JSON_6_1 class
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Class that encapsulates the processing of structures that adhere to the theme.json spec.
+ *
+ * This class is for internal core usage and is not supposed to be used by extenders (plugins and/or themes).
+ * This is a low-level API that may need to do breaking changes. Please,
+ * use get_global_settings, get_global_styles, and get_global_stylesheet instead.
+ *
+ * @access private
+ */
+class WP_Theme_JSON_6_1 extends WP_Theme_JSON_6_0 {
+
+	const ELEMENTS = array(
+		'link' => 'a',
+		'h1'   => 'h1',
+		'h2'   => 'h2',
+		'h3'   => 'h3',
+		'h4'   => 'h4',
+		'h5'   => 'h5',
+		'h6'   => 'h6',
+		// We have the .wp-block-button class so that this will target older buttons that have been serialized.
+		'button' => '.wp-element-button, .wp-block-button',
```

Comment:
Do we really want to support old markup? I understand the initial reasoning here but I wonder if we'll be stuck with this forever and there's also the question about why we don't support the "search" block for instance in the old markup.

Reply:
The problem is that if we don't do this we'll need to keep the CSS in the block theme so that old buttons that have already been serialised don't loose their appearance.

Reply:
The issue here is about adding a new theme to old content. If a theme implements button elements then it won't work on sites that were built using older versions of Gutenberg.

Reply:
Isn't that the same for old "search blocks"? I also expect that if a theme adds the element, he'd probably keep the old "core/button" style as well as a fallback maybe for these use-cases?

Reply:
No, because search blocks aren't serialized, so once the newer version of Gutenberg is released then the old markup will be gone. In the case of buttons I think we can expect old serialized buttons to be around for a long time into the future, which would mean all themes would need to keep the old CSS as a fallback.

Reply:
What happens if you define both element and core/button that are different? which one should take precedence?

Reply:
Global Styles outputs elements before blocks, so the button would win. That might change with https://github.com/WordPress/gutenberg/pull/40513

Reply:
> Global Styles outputs elements before blocks, so the button would win.

That makes sense to be honest :)

Reply:
I think I'm fine with this fallback class here personally. Thanks for the explanations.

Reply:
Thanks for asking the questions and listening to my answers!